core/src/main/scala/org/broadinstitute/dsde/rawls/metrics/RawlsInstrumented.scala

```diff
@@ -37,26 +43,48 @@ trait RawlsInstrumented extends DefaultInstrumented {
     */
   @implicitNotFound(msg = "Cannot expand instances of type ${A}")
   protected sealed trait Expansion[A] {
-    def makeName(key: String, a: A) = s"$key.${a.toString}"
+    def makeName(key: Option[String], a: A) = s"${getKey(key)}${a.toString}"
+    protected def getKey(k: Option[String]): String = k.map(k => s"$k.").getOrElse("")
   }
 
   // Typeclass instances:
 
   protected implicit object WorkspaceNameExpansion extends Expansion[WorkspaceName] {
     // Statsd doesn't allow slashes in metric names, so we override makeName to override
     // the default toString based implementation.
-    override def makeName(key: String, n: WorkspaceName): String = s"$key.${n.toString.replace('/', '.')}"
+    override def makeName(key: Option[String], n: WorkspaceName): String = s"${getKey(key)}${n.toString.replace('/', '.')}"
   }
 
   // Provide an implicit for UUIDs but can use the default makeName
   protected implicit object UUIDExpansion extends Expansion[UUID]
 
-  // Provide implicits for WorkflowStatus and SubmissionStatus using the default makeName.
+  // Provide implicits for various RawlsEnumerations using the default makeName.
+  // TODO: should we provide an expansion for RawlsEnumeration in general?
   //
   // These both take an upper type bound A <: WorkflowStatus|SubmissionStatus so they can
   // work either with the supertype (e.g. WorkflowStatus) or a subtype (e.g. WorkflowStatuses.Launching).
   protected implicit def WorkflowStatusExpansion[A <: WorkflowStatus] = new Expansion[A] {}
   protected implicit def SubmissionStatusExpansion[A <: SubmissionStatus] = new Expansion[A] {}
+  protected implicit def SubsystemExpansion[A <: Subsystem] = new Expansion[A] {}
+
+  protected implicit object HttpMethodExpansion extends Expansion[HttpMethod] {
+    override def makeName(key: Option[String], m: HttpMethod): String = s"${getKey(key)}${m.toString.toLowerCase}"
+  }
+
+  protected implicit object UriExpansion extends Expansion[Uri] {
+    override def makeName(key: Option[String], uri: Uri): String = s"${getKey(key)}${uri2String(uri)}"
+
+    private def uri2String(uri: Uri): String = {
+      val path = if (uri.path.startsWithSlash) uri.path.tail.toString else uri.path
+      path.toString.replace('/', '_')
```

Comment:
It's confusing in that it gives you several sub-metrics for the "uri-path" key. It doesn't look like SD supports sub-labels. What I'm trying is just enumerating them. E.g. given a path `/foo/bar/baz` I'm generating
```
label:uri-path-0 "foo"
label:uri-path-1 "bar"
label:uri-path-2 "baz"
```
Then users will see a drop-down for each path component where they can select "Any" or a specific value. I can see how that might be useful functionality, just need to make sure it's not too confusing.